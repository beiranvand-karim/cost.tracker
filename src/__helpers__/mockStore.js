import configureStore from "redux-mock-store";
import costItems from "./costItems";

const mockStore = configureStore();

export default mockStore({
  costItems: {
    costItems: costItems
  },
  sourceCostItems: {
    costItems: costItems
  }
});
