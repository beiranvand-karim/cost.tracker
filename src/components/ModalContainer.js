import { Modal, ModalBody, ModalHeader } from "reactstrap";
import CostItemForm from "./CostItemForm";
import React from "react";

export function ModalContainer({ modal, toggle }) {
  return (
    <Modal isOpen={modal} toggle={toggle}>
      <ModalHeader toggle={toggle}>Add A Cost</ModalHeader>
      <ModalBody>
        <CostItemForm toggle={toggle} />
      </ModalBody>
    </Modal>
  );
}
export default ModalContainer;
