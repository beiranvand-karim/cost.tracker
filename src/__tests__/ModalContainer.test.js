/* global expect, describe, it, beforeEach */
import ModalContainer from "../components/ModalContainer";
import mockStore from "../__helpers__/mockStore";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { shallow } from "enzyme";
import React from "react";

describe("<ModalContainer/>", () => {
  let store;
  const toggleMock = jest.fn();

  beforeEach(() => {
    store = mockStore;
  });

  it("should render without crashing", () => {
    shallow(<ModalContainer />);
  });

  it("should matches the snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ModalContainer toggle={toggleMock} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
