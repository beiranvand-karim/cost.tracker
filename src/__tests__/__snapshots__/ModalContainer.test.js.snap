// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalContainer/> should matches the snapshot 1`] = `null`;
